### DISCUSSION TITLE
Support for esphome

### DISCUSSION THREAD

**Discussion by tv4you2016** [schreibfaul1/ESP32-audioI2S#646] [Show and tell]
Hello @schreibfaul1 ,
noticed that esphome uses its library for audio reproduction (https://github.com/esphome/ESP32-audioI2S), but it is in an old version, I forked its lib and made the necessary changes to make it work in esphome which you can find here https://github.com/tv4you2016/ESP32-audioI2S-esphome

Are you interested in adding changes to your master code to directly support esphome?

**-DAUDIO_NO_SD_FS**  is the only flag used by them that makes the libs

```
#include <SPI.h>
#include <SD.h>
#include <SD_MMC.h>
#include <SPIFFS.h>
#include <FS.h>
#include <FFat.h>
```

are not added, this also leads to some small changes in the code.
These changes were tested by me and I can say that it works up to version v3.0.8d1

**Comment by schreibfaul1** [comment]
I'm not familiar with esphome, I've never used it. The libraries are all from the Arduino ESP32 framework. Currently the objects such as SD or SPIFFS are passed by reference and therefore the "includes" are no longer required. I deleted all but not FS.h, which is essential.
You can please test this once, maybe it will work like this in esphome.

**Reply by tv4you2016** [reply]
I just tested it with your latest commit (5043901), but once I had to add define as you can check (https://github.com/tv4you2016/ESP32-audioI2S-esphome) for it to work... I can say that it works perfectly With my change in esphome and plaformio I also gain 1% of memory :D. Let's say the only change was to add
 #include <FS.h> selectable in compilation
